### PR TITLE
Publishing releases draft

### PR DESCRIPTION
**Pull request by dlew**
An attempt at getting us back to a releasable state.

The bintray/maven configuration values are derived from the pre-existing releases + what defaults can be gleaned from the previously used [gradle-rxjava-project-plugin](https://github.com/nebula-plugins/gradle-rxjava-project-plugin).

This is a draft for a few reasons:

1. I'm not sure that the values are correct.
1. I don't have API access so I can't directly test whether it actually works.
1. `bintray` has `dryRun = true` at the moment, so I won't accidentally break anything.

Suggestions on how to improve/fix are more than welcome!

### CODE REVIEW

File: gradle/artifacts.gradle [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+            classifier = "${classifierPrefix}javadoc"
+            from javadocTask.destinationDir
+        }
+
+        Task sourcesJarTask = task("generate${variant.name.capitalize()}SourcesJar", type: Jar) {
+            group = 'artifact'
+            description = "Generates sources jar for $variant.name"
+
+            classifier = "${classifierPrefix}sources"
+            from variant.javaCompiler.source
```
Comment (roman-mazur) [inline gradle/artifacts.gradle]: Should not this be `javaCompile.source`?

File: build.gradle [hunk 2]
```diff
@@ -13,6 +13,21 @@ ext {
...
+    POM_DESCRIPTION= "RxAndroid"
+
+    POM_URL="https://github.com/ReactiveX/RxAndroid"
+    POM_SCM_URL="https://github.com/ReactiveX/RxAndroid"
+    POM_SCM_CONNECTION="scm:git:https://github.com/ReactiveX/RxAndroid.git"
+    POM_SCM_DEV_CONNECTION="scm:git:git@github.com:ReactiveX/RxAndroid.git"
+
+    POM_LICENCE_NAME="The Apache Software License, Version 2.0"
+    POM_LICENCE_URL="http://www.apache.org/licenses/LICENSE-2.0.txt"
+    POM_LICENCE_DIST="repo"
```
Comment (mttkay) [inline build.gradle]: could we pull these into a `.properties` file?

File: gradle/artifacts.gradle (hunk 1 above)
Comment (dlew) [inline gradle/artifacts.gradle]: No. `variant.java.compile` is deprecated. From [BaseVariant.java](https://android.googlesource.com/platform/tools/base/+/master/build-system/gradle-core/src/main/groovy/com/android/build/gradle/api/BaseVariant.java):

```java
/**
 * Returns the Java Compilation task if javac was configured to compile the source files.
 * @deprecated prefer {@link #getJavaCompiler} which always return the java compiler task
 * irrespective of which tool chain (javac or jack) used.
 */
@Nullable
@Deprecated
JavaCompile getJavaCompile() throws IllegalStateException;

/**
 * Returns the Java Compiler task which can be either javac or jack depending on the project
 * configuration.
 */
@NonNull
AbstractCompile getJavaCompiler();
```

File: build.gradle (hunk 2 above)
Comment (dlew) [inline build.gradle]: Could call - stuffed them into `gradle.properties`.

File: gradle/artifacts.gradle (hunk 1 above)
Comment (roman-mazur) [inline gradle/artifacts.gradle]: Ah, indeed. Jack is coming :)

File: rxandroid/build.gradle [hunk 3]
```diff
@@ -26,3 +41,76 @@ dependencies {
...
+
+// Configure gradle-bintray-plugin
+bintray {
+    configurations = ['archives']
+
+    // TODO: DELETE ME WHEN READY!
+    dryRun = true
+
+    pkg {
+        repo = 'RxJava'
```
Comment (JakeWharton) [inline rxandroid/build.gradle]: Why is this RxJava and not RxAndroid?

Comment (dlew) [inline rxandroid/build.gradle]: As far as I could tell, this was the data sent by the rxjava nebula plugin: https://github.com/nebula-plugins/gradle-rxjava-project-plugin/blob/gradle-2.2/src/main/groovy/nebula/plugin/rxjavaproject/RxJavaPublishingPlugin.groovy

**Comment by JakeWharton** [comment]
Any reason not to merge?

**Comment by dlew** [comment]
I'm on vacation and can't fix anything until Sunday, if that's a reason.

**Comment by dlew** [comment]
I'm back now.

I'm still not confident the code is correct (since I could never actually test it out); only someone with the keys to the kingdom could do that I think. But I'm fairly sure this is a step in the right direction.

**Comment by dlew** [comment]
I've added artifactory support for snapshots.

Again, without credentials I can't test the code directly - but theoretically if this were merged then travis would upload a snapshot.

**Pull request merged**